shapely/geometry/linestring.py

```diff
@@ -40,7 +42,9 @@ def __init__(self, coordinates=None):
         """
         BaseGeometry.__init__(self)
         if coordinates is not None:
-            self._set_coords(coordinates)
+            ret = geos_linestring_from_py(coordinates)
+            if ret is not None:
+                self._geom, self._ndim = ret
```

Comment:
@jorisvandenbossche I appreciate how we will be able to just delete _set_coords in the future. Thank you!